### ISSUE TITLE
Option 1: Daml Sandbox - Invalid dar file

### ISSUE THREAD

**Issue by Imchrisrivers** [digital-asset/ex-cdm-swaps#33]
After running the 'daml build' and '(cd app; sbt compile)', for the Option 1 implementation, I receive an "invalid dar file: CdmSwaps-1.0.0.dar" ExitFailure 1 error after running this command: daml sandbox .daml/dist/CdmSwaps-1.0.0.dar. 

The raw command: java -jar /Users/craniac/.daml/sdk/0.13.22/sandbox/sandbox.jar .daml/dist/CdmSwaps-1.0.0.dar

**Comment by GeorgSchneider** [comment]
Are you running the `daml sandbox .daml/dist/CdmSwaps-1.0.0.dar` command from the `app` directory? You need to run this from the root dir in the repo.

**Comment by Imchrisrivers** [comment]
Yes, I run the daml sandbox command after I CD into App folder.

Please see attached screenshot:

[cid:8779E6EB-B5E2-4546-89F7-E14028683D78]

Best,
Chris

CHRIS RIVERS
CEO | Optimis, Inc.
222 South Church Street, Suite 100 |
Charlotte, NC 28202
M: + 1 (704) 565-9704
optimisinc.co<http://optimisinc.co>





On Feb 3, 2020, at 5:42 AM, Georg Schneider <notifications@github.com<mailto:notifications@github.com>> wrote:


Are you running the daml sandbox .daml/dist/CdmSwaps-1.0.0.dar command from the app directory? You need to run this from the root dir in the repo.

**Comment by GeorgSchneider** [comment]
You need to run the sandbox command from the root repo folder.

**Comment by Imchrisrivers** [comment]
After running the daml build and (cd app; sbt compile) commands, I cd back into the root folder and ran the daml sandbox command successfully within the same shell. (Might I suggest the root folder instruction be included on Git?)

I opened a second shell for daml navigator server command and ran this successfully.

I opened another shell for REPL command. This too was run successfully.

I opened the last shell for the automation command and got this error:

[cid:509F73EF-6215-4A24-9BBE-C1D7F03D9B82]


CHRIS RIVERS
CEO | Optimis, Inc.
222 South Church Street, Suite 100 |
Charlotte, NC 28202
M: + 1 (704) 565-9704
optimisinc.co<http://optimisinc.co>





On Feb 4, 2020, at 2:20 AM, Georg Schneider <notifications@github.com<mailto:notifications@github.com>> wrote:


You need to run the sandbox command from the root repo folder.

**Comment by Imchrisrivers** [comment]
Further to my below, upon closer inspection, I receive an out of memory error when running the daml navigator server command. It appears that the 16 GB hardware limit is needed for both Options 1 and 2…?


CHRIS RIVERS
CEO | Optimis, Inc.
222 South Church Street, Suite 100 |
Charlotte, NC 28202
M: + 1 (704) 565-9704
optimisinc.co<http://optimisinc.co>





On Feb 4, 2020, at 6:12 PM, Christopher Rivers <chris@optimisinc.co<mailto:chris@optimisinc.co>> wrote:

After running the daml build and (cd app; sbt compile) commands, I cd back into the root folder and ran the daml sandbox command successfully within the same shell. (Might I suggest the root folder instruction be included on Git?)

I opened a second shell for daml navigator server command and ran this successfully.

I opened another shell for REPL command. This too was run successfully.

I opened the last shell for the automation command and got this error:

<Screen Shot 2020-02-04 at 6.11.43 PM.png>


CHRIS RIVERS
CEO | Optimis, Inc.
222 South Church Street, Suite 100 |
Charlotte, NC 28202
M: + 1 (704) 565-9704
optimisinc.co<http://optimisinc.co/>





On Feb 4, 2020, at 2:20 AM, Georg Schneider <notifications@github.com<mailto:notifications@github.com>> wrote:


You need to run the sandbox command from the root repo folder.

**Comment by GeorgSchneider** [comment]
Yes, the CDM model is quite large, so it could be that it requires >8gb to run navigator. I haven't tested the exact requirements yet for this option.

**Comment by Imchrisrivers** [comment]
Ok understood. I look forward to your feedback.

I’ve considered deploying the docker container in the cloud. Have you consider that as a 3rd option? It would increase the usability of the app (and may be a quicker workaround than reworking code changes)

Best,
Chris

Christopher Rivers
CEO | Optimis, Inc.
222 South Church Street, Suite 100 |
Charlotte, NC 28202
M: +1 (704) 565-9704<tel:+1%20(704)%20565-9704>
optimisinc.co<http://optimisinc.co/>

On Feb 10, 2020, at 8:17 AM, Georg Schneider <notifications@github.com> wrote:

﻿

Yes, the CDM model is quite large, so it could be that it requires >8gb to run navigator. I haven't tested the exact requirements yet for this option.
